src/plugins/ui/edition.ts

```diff
@@ -442,12 +446,8 @@ export class EditionPlugin extends UIPlugin {
     }
   }
 
-  private cancelEdition() {
-    if (this.mode === "inactive") {
-      return;
-    }
-    this.mode = "inactive";
-    this.selection.release(this);
+  private cancelEditionAndActivateSheet() {
+    this.cancelEdition();
```

Comment:
when the mode is inactive, this function should also return early. Again a inssue of design ...
I think taht's the main reason you have failing tests ;-)